### PR TITLE
env_process: added check host component version functionality V2

### PR DESCRIPTION
**Pull request by hereischen**
It enabled using required_kernel and required_qemu to ensure
the test runs only on certain versions.

id: 1519710
Signed-off-by: Haotong Chen hachen@redhat.com

### CODE REVIEW

**Comment by hereischen** [comment]
v1: (https://github.com/avocado-framework/avocado-vt/pull/1250)
Add a VersionInterval Class.

**Comment by hereischen** [comment]
eg:
in boot.cfg:
` required_qemu = [2.9.0,2.9.1)`
`required_kernel = [3.0,)`

CANCEL 1-smp_4.4096m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.boot -> TestCancel: From host, get qemu version 2.10.0-17, which is not in [2.9.0,2.9.1)

**Comment by hereischen** [comment]
@luckyh Many thanks for your review.
@pingl2016 @ldoktor please help review, thanks.

**Comment by hereischen** [comment]
A note form @pingl2016 "Once the patch is merged. Please backport the patch to internal tree for rhel6. And notice that 'VirtTest' object has no attribute 'cancel' for rhel6 branch. Thanks"

File: virttest/env_process.py [hunk 1]
```diff
@@ -765,6 +766,19 @@ def preprocess(test, params, env):
     logging.debug("KVM userspace version: %s" % kvm_userspace_version)
     test.write_test_keyval({"kvm_userspace_version": kvm_userspace_version})
 
+    # Checking required qemu, if not satisfied, cancel test
```
Comment (xutian) [inline virttest/env_process.py]: it looks duplicate with above changes,  please try to remove duplicate.

Comment (hereischen) [reply virttest/env_process.py]: Many thanks for the review.
There is a plan to re-design the check procedures in `preprocess` and `postprocess`, this is sort like a temporary implemantation  to check the version of qemu and kernel.
There will be a new patch to handle this.

**Comment by pingl2016** [comment]
Test passed as below

**Pull request merged**